src/Runtime/Scripts/cshtml5.js

```diff
@@ -325,6 +334,158 @@ document.errorCallback = function (error, IndexOfNextUnmodifiedJSCallInList) {
+        }
+    }                                      
+    document.ondblclick = function(e) {   
+       if(e.doNotReroute == undefined)
+        {
+               var element = document.getElementByIdSafe(id);
+               document.reroute(e, element);
+        }
+    }
+}
+
+document.setVisualBounds = function(id, left, top, width, height, bSetAbsolutePosition, bSetZeroMargin, bSetZeroPadding) {
+    var element = document.getElementByIdSafe(id);
+    if (element)
+    {
+        element.style.left = left + "px";
+        element.style.top = top + "px";
+        element.style.width = width + "px";
+        element.style.height = height + "px";
+        
+        if (bSetAbsolutePosition) {
+            element.style.position = "absolute";
+        }
+        if (bSetZeroMargin) {
+            element.style.margin = "0";
+        }
+        if (bSetZeroPadding) {
+            element.style.padding = "0";
+        }
+    }
+}
+
+document.setPosition = function(id, left, top, bSetAbsolutePosition, bSetZeroMargin, bSetZeroPadding) {
+    var element = document.getElementByIdSafe(id);
+    if (element)
+    {
+        element.style.left = left + "px";
+        element.style.top = top + "px";
+        
+        if (bSetAbsolutePosition) {
+            element.style.position = "absolute";
+        }
+        if (bSetZeroMargin) {
+            element.style.margin = "0";
+        }
+        if (bSetZeroPadding) {
+            element.style.padding = "0";
+        }
+    }
+}
+
+document.measureTextBlock = function(text, fontSize, fontFamily, fontStyle, fontWeight, textWrapping, padding, width, maxWidth) {
+    var element = document.measureTextBlockElement;
+    if (element)
+    {
+        var runElement = element.firstElementChild;
+        if (runElement != null) {
+            runElement.innerText = text;
+            runElement.style.fontSize = fontSize;
+            runElement.style.fontWeight = fontWeight;
+        }
+
+        if (fontSize.length > 0) {
+            element.style.fontSize = fontSize;
+        }
+        if (fontFamily.length > 0) {
+            if (fontFamily === "-") {
+                fontFamily = "";
+            }
+            element.style.fontFamily = fontFamily;
+        }
+        if (fontStyle.length > 0) {
+            element.style.fontStyle = fontStyle;
+        }
+        if (fontWeight.length > 0) {
+            element.style.fontWeight = fontWeight;
+        }
+        if (textWrapping.length > 0) {
+            element.style.whiteSpace = textWrapping;
+        }
+        if (padding.length > 0) {
+            element.style.boxSizing = "border-box";
+            element.style.padding = padding;
+        }
+
+        element.style.width = width;
+        element.style.maxWidth = maxWidth;
+
+        return element.offsetWidth + "|" + element.offsetHeight;
+    }
+
+    return "0|0";
+}
+
+document.setContentString = function(id, text, removeTextWrapping) {
+    var el = document.getElementByIdSafe(id);

```

Comment:
do not use 'getElementByIdSafe' in js scripts
```suggestion
    var el = document.getElementById(id);
```